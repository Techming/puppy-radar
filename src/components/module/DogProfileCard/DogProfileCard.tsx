import React from "react";
import { Box, Flex, Image, Text } from "@chakra-ui/react";
import {
  faHeart,
  faLocationDot,
  faPaw,
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

interface DogProfileCardProps {
  id: string;
  img: string;
  name: string;
  age: number;
  zip_code: string;
  breed: string;
  liked: boolean;
  onToggleLike: () => void;
}

const DogProfileCard = (props: DogProfileCardProps) => {
  return (
    <Flex
      position={"relative"}
      w={"250px"}
      h={"300px"}
      bg={"gray.100"}
      borderRadius={"10px"}
      overflow={"hidden"}
      filter={
        props.liked
          ? "drop-shadow(0px 2px 12px rgba(190, 46, 0, 0.8))"
          : "drop-shadow(0px 2px 10px rgba(0, 0, 0, 0.25))"
      }
      _hover={{
        filter: props.liked
          ? "drop-shadow(0px 2px 16px rgba(190, 46, 0, 0.8))"
          : "drop-shadow(0px 2px 16px rgba(0, 0, 0, 0.5))",
      }}
      transition={"filter 150ms ease-in-out"}
    >
      <Box
        position={"absolute"}
        top={"17px"}
        right={"15px"}
        w={"30px"}
        h={"30px"}
        color={props.liked ? "#E83333" : "gray.300"}
        transition={"color 150ms ease-in-out"}
        filter={"drop-shadow(0px 0px 6px rgba(0, 0, 0, 0.4))"}
        zIndex={3}
        cursor={"pointer"}
        aria-label="Like this dog."
        onClick={props.onToggleLike}
      >
        <FontAwesomeIcon icon={faHeart} height={"30px"} />
      </Box>
      <Flex
        flexDir={"column"}
        justifyContent={"flex-end"}
        position={"absolute"}
        top={0}
        left={0}
        paddingX={"15px"}
        paddingY={"25px"}
        width={"100%"}
        height={"100%"}
        bg={
          "linear-gradient(180deg, rgba(0, 0, 0, 0) 45%, rgba(0, 0, 0, 0.9) 100%)"
        }
        color={"white"}
        borderRadius={"10px"}
        zIndex={2}
      >
        <Text fontWeight={"bold"} fontSize={"xl"} mb={1}>
          {props.name}
        </Text>
        <Flex mb={1}>
          <Text w={"50%"}>{props.age} YO</Text>
          <Flex alignItems={"center"}>
            <FontAwesomeIcon icon={faLocationDot} height={"16px"} />
            <Text ml={2}>{props.zip_code}</Text>
          </Flex>
        </Flex>
        <Flex alignItems={"center"}>
          <FontAwesomeIcon icon={faPaw} height={"16px"} />
          <Text ml={2}>{props.breed}</Text>
        </Flex>
      </Flex>
      <Image
        w={"250px"}
        h={"300px"}
        src={props.img}
        alt={`A dog named${props.name}.`}
        objectFit={"cover"}
        zIndex={1}
      />
    </Flex>
  );
};

export default DogProfileCard;
