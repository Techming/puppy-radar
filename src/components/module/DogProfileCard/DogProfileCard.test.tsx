import * as React from "react";

import { getByText, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import DogProfileCard from "./DogProfileCard";

describe("DogProfileCard", () => {
  const FAKE_DOG_PROPS = {
    id: "fake_id",
    img: "https://www.google.com",
    name: "fake dog name",
    age: 2,
    zip_code: "12345",
    breed: "Some Fake Breed",
    liked: false,
    onToggleLike: () => {},
  };
  it("renders correctly", () => {
    const dogProfileCard = render(<DogProfileCard {...FAKE_DOG_PROPS} />);

    expect(dogProfileCard).toMatchSnapshot();
  });

  it("shows all the information feeded but id", () => {
    const { container } = render(<DogProfileCard {...FAKE_DOG_PROPS} />);

    expect(getByText(container, FAKE_DOG_PROPS.name)).toBeVisible();
    expect(getByText(container, FAKE_DOG_PROPS.age + " YO")).toBeVisible();
    expect(getByText(container, FAKE_DOG_PROPS.zip_code)).toBeVisible();
    expect(getByText(container, FAKE_DOG_PROPS.breed)).toBeVisible();
  });
});
