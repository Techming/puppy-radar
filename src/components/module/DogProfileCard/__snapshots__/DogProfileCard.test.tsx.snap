// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogProfileCard renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1vp3oai"
      >
        <div
          aria-label="Like this dog."
          class="css-1k0nwcz"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            height="30px"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="css-1ky8hpg"
        >
          <p
            class="chakra-text css-1leha7a"
          >
            fake dog name
          </p>
          <div
            class="css-he2swc"
          >
            <p
              class="chakra-text css-11ze7cv"
            >
              2
               YO
            </p>
            <div
              class="css-70qvj9"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-location-dot "
                data-icon="location-dot"
                data-prefix="fas"
                focusable="false"
                height="16px"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M215.7 499.2C267 435 384 279.4 384 192C384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2c12.3 15.3 35.1 15.3 47.4 0zM192 128a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
                  fill="currentColor"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                12345
              </p>
            </div>
          </div>
          <div
            class="css-70qvj9"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-paw "
              data-icon="paw"
              data-prefix="fas"
              focusable="false"
              height="16px"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M226.5 92.9c14.3 42.9-.3 86.2-32.6 96.8s-70.1-15.6-84.4-58.5s.3-86.2 32.6-96.8s70.1 15.6 84.4 58.5zM100.4 198.6c18.9 32.4 14.3 70.1-10.2 84.1s-59.7-.9-78.5-33.3S-2.7 179.3 21.8 165.3s59.7 .9 78.5 33.3zM69.2 401.2C121.6 259.9 214.7 224 256 224s134.4 35.9 186.8 177.2c3.6 9.7 5.2 20.1 5.2 30.5v1.6c0 25.8-20.9 46.7-46.7 46.7c-11.5 0-22.9-1.4-34-4.2l-88-22c-15.3-3.8-31.3-3.8-46.6 0l-88 22c-11.1 2.8-22.5 4.2-34 4.2C84.9 480 64 459.1 64 433.3v-1.6c0-10.4 1.6-20.8 5.2-30.5zM421.8 282.7c-24.5-14-29.1-51.7-10.2-84.1s54-47.3 78.5-33.3s29.1 51.7 10.2 84.1s-54 47.3-78.5 33.3zM310.1 189.7c-32.3-10.6-46.9-53.9-32.6-96.8s52.1-69.1 84.4-58.5s46.9 53.9 32.6 96.8s-52.1 69.1-84.4 58.5z"
                fill="currentColor"
              />
            </svg>
            <p
              class="chakra-text css-14gt3bc"
            >
              Some Fake Breed
            </p>
          </div>
        </div>
        <img
          alt="A dog namedfake dog name."
          class="chakra-image css-1o3he7n"
          src="https://www.google.com"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1vp3oai"
    >
      <div
        aria-label="Like this dog."
        class="css-1k0nwcz"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart "
          data-icon="heart"
          data-prefix="fas"
          focusable="false"
          height="30px"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="css-1ky8hpg"
      >
        <p
          class="chakra-text css-1leha7a"
        >
          fake dog name
        </p>
        <div
          class="css-he2swc"
        >
          <p
            class="chakra-text css-11ze7cv"
          >
            2
             YO
          </p>
          <div
            class="css-70qvj9"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-location-dot "
              data-icon="location-dot"
              data-prefix="fas"
              focusable="false"
              height="16px"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M215.7 499.2C267 435 384 279.4 384 192C384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2c12.3 15.3 35.1 15.3 47.4 0zM192 128a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
                fill="currentColor"
              />
            </svg>
            <p
              class="chakra-text css-14gt3bc"
            >
              12345
            </p>
          </div>
        </div>
        <div
          class="css-70qvj9"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-paw "
            data-icon="paw"
            data-prefix="fas"
            focusable="false"
            height="16px"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M226.5 92.9c14.3 42.9-.3 86.2-32.6 96.8s-70.1-15.6-84.4-58.5s.3-86.2 32.6-96.8s70.1 15.6 84.4 58.5zM100.4 198.6c18.9 32.4 14.3 70.1-10.2 84.1s-59.7-.9-78.5-33.3S-2.7 179.3 21.8 165.3s59.7 .9 78.5 33.3zM69.2 401.2C121.6 259.9 214.7 224 256 224s134.4 35.9 186.8 177.2c3.6 9.7 5.2 20.1 5.2 30.5v1.6c0 25.8-20.9 46.7-46.7 46.7c-11.5 0-22.9-1.4-34-4.2l-88-22c-15.3-3.8-31.3-3.8-46.6 0l-88 22c-11.1 2.8-22.5 4.2-34 4.2C84.9 480 64 459.1 64 433.3v-1.6c0-10.4 1.6-20.8 5.2-30.5zM421.8 282.7c-24.5-14-29.1-51.7-10.2-84.1s54-47.3 78.5-33.3s29.1 51.7 10.2 84.1s-54 47.3-78.5 33.3zM310.1 189.7c-32.3-10.6-46.9-53.9-32.6-96.8s52.1-69.1 84.4-58.5s46.9 53.9 32.6 96.8s-52.1 69.1-84.4 58.5z"
              fill="currentColor"
            />
          </svg>
          <p
            class="chakra-text css-14gt3bc"
          >
            Some Fake Breed
          </p>
        </div>
      </div>
      <img
        alt="A dog namedfake dog name."
        class="chakra-image css-1o3he7n"
        src="https://www.google.com"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
